backend/authentication/auth_routes.js

```diff
@@ -1,11 +1,17 @@
 const express = require("express");
 const router = express.Router();
 const authController = require("./auth_controller");
+const authMiddleware = require("./auth_middleware");
 
 router.post("/register", authController.register);
 router.post("/login", authController.login);
 router.get("/github", authController.githubAuth);
 router.get("/github/callback", authController.githubCallback);
 router.post("/github/logout", authController.logout);
+router.get("/check", authMiddleware, (req, res) => {
+  res.status(200).json({
+    message: "Authenticated",
+  });
+});
```

Comment:
It is to verify that the user is still authenticated in the authenticatedContext file

Reply:
But what is the intention of this, are expecting to be calling this route multiple times during the usage of the site or just to verify the user is still logged when they access or something else?

I'm asking this because I feel that this route might be a little redundant, if the user already has a token then it means that it's already logged in. If the authMiddleware fails in any route that needs it then it means that there is no token, is wrong or is expired which in all cases should fail and you can handle a redirect to /login in the frontend